src/bid/encoding.rs

```diff
@@ -62,72 +68,70 @@ impl Into<StorageScalar> for Bid {
     }
 }
 
-impl Bid {
-    /// Applies a preimage_gadget to the `StorageBid` fields hashing them and
-    /// constraining the result of the sponge hash to the real/expected
-    /// `StorageBid` encoded value expressed as `Scalar/StorageScalar`.
-    ///
-    /// The expected encoded value is a Public Input for this circuit.
-    /// Appart from that, the preimage_gadget hashing result is returned.
-    pub(crate) fn preimage_gadget(
-        &self,
-        composer: &mut StandardComposer,
-    ) -> Variable {
-        // This field represents the types of the inputs and has to be the same
-        // as the default one.
-        // It has been already checked that it's safe to unwrap here since the
-        // value fits correctly in a `BlsScalar`.
-        let type_fields = BlsScalar::from_bytes(&TYPE_FIELDS).unwrap();
-
-        // Add to the composer the values required for the preimage.
-        let mut messages: Vec<Variable> = vec![];
-        messages.push(composer.add_input(type_fields));
-        // Push both JubJubAffine coordinates as a Scalar.
-        self.encrypted_data.cipher().iter().for_each(|c| {
-            let c = composer.add_input(*c);
-            messages.push(c);
-        });
-
-        // Push both JubJubAffine coordinates as a Scalar.
-        messages.push(composer.add_input(self.stealth_address.pk_r().get_x()));
-        messages.push(composer.add_input(self.stealth_address.pk_r().get_y()));
-        // Push both JubJubAffine coordinates as a Scalar.
-        messages.push(composer.add_input(self.stealth_address.R().get_x()));
-        messages.push(composer.add_input(self.stealth_address.R().get_y()));
-        messages.push(composer.add_input(self.hashed_secret));
-        // Push both JubJubAffine coordinates as a Scalar.
-        messages.push(composer.add_input(self.c.get_x()));
-        messages.push(composer.add_input(self.c.get_y()));
-        // Add elebility & expiration timestamps.
-        messages.push(composer.add_input(self.elegibility_ts));
-        messages.push(composer.add_input(self.expiration_ts));
-
-        // Perform the sponge_hash inside of the Constraint System
-        let storage_bid_hash = sponge_hash_gadget(composer, &messages);
-        // Constraint the hash to be equal to the real one
-        let real_hash: StorageScalar = self.clone().into();
-        let real_hash: BlsScalar = real_hash.into();
-        composer.constrain_to_constant(
-            storage_bid_hash,
-            BlsScalar::zero(),
-            -real_hash,
-        );
-        storage_bid_hash
+impl Into<StorageScalar> for Bid {
+    fn into(self) -> StorageScalar {
+        (&self).into()
     }
 }
 
+/// Hashes the internal Bid parameters using the Poseidon252 hash
+/// function and the cannonical encoding for hashing returning a
+/// Variable which contains the hash of the Bid.
+pub(crate) fn preimage_gadget(
+    composer: &mut StandardComposer,
+    encrypted_data: (Variable, Variable),
```

Comment:
This assumption of the poseidon cipher as a representation of two variables is not the responsibility of blind bid.

We should leave a TODO note to update to a circuit representation of the cipher when we have it available on the poseidon library.

Reply:
Hmm not sure I'm following. WDYM? It's part of the protocol knowing that we just have two encrypted fields inside.

Reply:
What @vlopes11 says is that the fact that Poseidon Cipher representation is just two variables is an implementation details that is part of the poseidon lib.
(For example n Poseidon most of the time is returned an array of fixed size – configured by constants)
I will just check with @vlopes11 what a reasonable TODO (with example) would be here and then just add that.

Reply:
Will add the TODO and merge. Although idk if we will really adress this that way.